src/spatio_temporal_voxel_grid.cpp

```diff
@@ -330,12 +341,38 @@ double SpatioTemporalVoxelGrid::GetFrustumAcceleration( \
 }
 
 /*****************************************************************************/
-void SpatioTemporalVoxelGrid::GetOccupancyPointCloud( \
-                                       pcl::PointCloud<pcl::PointXYZ>::Ptr& pc)
+void SpatioTemporalVoxelGrid::GetOccupancyPointCloud(\
+                                       sensor_msgs::PointCloud2::Ptr& pc2)
 /*****************************************************************************/
 {
-  // return the pointcloud stored
-  pc = _pc;
+  // convert the grid points stored in a PointCloud2
+  sensor_msgs::PointCloud2::Ptr grid_points_cloud (new sensor_msgs::PointCloud2());
+  grid_points_cloud->width  = _grid_points->size();
+  grid_points_cloud->height = 1;
+  grid_points_cloud->is_dense = true;
+
+  sensor_msgs::PointCloud2Modifier modifier(*grid_points_cloud);
+
+  modifier.setPointCloud2Fields(3,
+                                "x", 1, sensor_msgs::PointField::FLOAT32,
+                                "y", 1, sensor_msgs::PointField::FLOAT32,
+                                "z", 1, sensor_msgs::PointField::FLOAT32);
+  modifier.setPointCloud2FieldsByString(1, "xyz");
+
+  sensor_msgs::PointCloud2Iterator<float>iter_x(*grid_points_cloud, "x");
+  sensor_msgs::PointCloud2Iterator<float>iter_y(*grid_points_cloud, "y");
+  sensor_msgs::PointCloud2Iterator<float>iter_z(*grid_points_cloud, "z");
+
+  for(std::list<geometry_msgs::Point32>::iterator it = _grid_points->begin(); it != _grid_points->end(); ++it){
+    const geometry_msgs::Point32& pt = *it;
+    *iter_x = pt.x;
+    *iter_y = pt.y;
+    *iter_z = pt.z;
+    ++iter_x; ++iter_y; ++iter_z;
+  }
+
+  // return the pointcloud
+  pc2 = grid_points_cloud;
```

Comment:
Why don't you use the already allocated `pc2` throughout the function?

Reply:
Good point